### PR TITLE
Update docker

### PR DESCRIPTION
**Pull request by jacob-delgado**

### CODE REVIEW

File: docker/build-tools/Dockerfile [hunk 1]
```diff
@@ -482,8 +482,8 @@ FROM ubuntu:jammy as base_os_context
 ENV DEBIAN_FRONTEND=noninteractive
 
 ENV CONTAINERD_VERSION=1.6.28-1
-ENV DOCKER_VERSION=5:24.0.9-1~ubuntu.22.04~jammy
-ENV DOCKER_BUILDX_VERSION=0.11.2-1~ubuntu.22.04~jammy
+ENV DOCKER_VERSION=5:25.0.2-1~debian.12~bookworm
+ENV DOCKER_BUILDX_VERSION=0.12.1-1~debian.12~bookworm
```
Comment (jacob-delgado) [inline docker/build-tools/Dockerfile:486]:
```suggestion
ENV DOCKER_VERSION=5:25.0.2-1~debian.12~jammy
ENV DOCKER_BUILDX_VERSION=0.12.1-1~debian.12~jammy
```

File: docker/build-tools/Dockerfile [hunk 2]
```diff
@@ -482,8 +482,8 @@ FROM ubuntu:jammy as base_os_context
 ENV DEBIAN_FRONTEND=noninteractive
 
 ENV CONTAINERD_VERSION=1.6.28-1
-ENV DOCKER_VERSION=5:24.0.9-1~ubuntu.22.04~jammy
-ENV DOCKER_BUILDX_VERSION=0.11.2-1~ubuntu.22.04~jammy
+ENV DOCKER_VERSION=5:25.0.2-1~debian.12~jammy
+ENV DOCKER_BUILDX_VERSION=0.12.1-1~debian.12~jammy
```
Comment (jacob-delgado) [inline docker/build-tools/Dockerfile:486]:
```suggestion
ENV DOCKER_VERSION=5:25.0.2-1~ubuntu.22.04~jammy
ENV DOCKER_BUILDX_VERSION=0.12.1-1~ubuntu.22.04~jammy
```

File: docker/build-tools/Dockerfile [hunk 3]
```diff
@@ -542,6 +542,7 @@ RUN apt-key add /tmp/docker-key
 ARG TARGETARCH
 RUN add-apt-repository "deb [arch=${TARGETARCH}] https://download.docker.com/linux/ubuntu $(lsb_release -sc) stable"
 RUN apt-get update
+RUN sed -i 's/ulimit -Hn/# ulimit -Hn/g' /etc/init.d/docker
 RUN apt-get -y install --no-install-recommends docker-ce="${DOCKER_VERSION}" docker-ce-cli="${DOCKER_VERSION}" containerd.io="${CONTAINERD_VERSION}" docker-buildx-plugin="${DOCKER_BUILDX_VERSION}"
```
Comment (dhawton) [inline docker/build-tools/Dockerfile:546]:
```suggestion
RUN apt-get -y install --no-install-recommends docker-ce="${DOCKER_VERSION}" docker-ce-cli="${DOCKER_VERSION}" containerd.io="${CONTAINERD_VERSION}" docker-buildx-plugin="${DOCKER_BUILDX_VERSION}"
RUN sed -i 's/ulimit -Hn/# ulimit -Hn/g' /etc/init.d/docker
```

**Comment by jacob-delgado** [comment]
Need update to kind first https://github.com/kubernetes-sigs/kind/releases/tag/v0.21.0

**Review by dhawton — APPROVED** [review]
LGTM. Leaving DNM for @jacob-delgado to verify but kind was merged already

**Review by ericvn — APPROVED** [review]

**Review by kfaseela — APPROVED** [review]

**Pull request merged**